packet/conn.go

```diff
@@ -97,10 +96,30 @@ func (c *Conn) ReadPacketReuseMem(dst []byte) ([]byte, error) {
 
 	if err := c.ReadPacketTo(buf); err != nil {
 		return nil, errors.Trace(err)
+	}
+
+	readBytes := buf.Bytes()
+	readSize := len(readBytes)
+	var result []byte
+	if len(dst) > 0 {
```

Comment:
Maybe it will be enough for us leave the old version
```go
result = append(dst, buf.Bytes()...)
```
but change
```go
defer utils.BytesBufferPut(buf)
```
to
```go
defer func(){
	if readSize <= utils.TooBigBlockSize { // Or move this logic into utils.BytesBufferPut itself?
		utils.BytesBufferPut(buf)
	}
}()
```